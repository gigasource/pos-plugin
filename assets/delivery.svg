<svg width="45" height="42" viewBox="0 0 45 42" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path fill-rule="evenodd" clip-rule="evenodd" d="M11.879 0C10.9836 0 10.1973 0.595118 9.95416 1.45686L0.844222 33.7414C0.534042 34.8406 1.20586 35.9759 2.31869 36.2331L25.2431 41.5315C25.5493 41.6023 25.8678 41.5999 26.1729 41.5245L42.7095 37.4413C43.78 37.177 44.4346 36.0961 44.1729 35.025L35.6154 0H11.879Z" fill="#FBB148"/>
    <path fill-rule="evenodd" clip-rule="evenodd" d="M25.7783 41.5517L44.0359 36.9251L35.3371 0L25.7783 41.5517Z" fill="#E49337"/>
    <path fill-rule="evenodd" clip-rule="evenodd" d="M34.9548 32.2477L44.037 36.5152L34.9082 0L34.9548 32.2477Z" fill="#E9AD2E"/>
    <path fill-rule="evenodd" clip-rule="evenodd" d="M25.7783 41.5517L36.2112 32.6097L36.1631 0L25.7783 41.5517Z" fill="#A56630"/>
    <path fill-rule="evenodd" clip-rule="evenodd" d="M27.6937 7.64187L14.7489 8.92306L7.10714 7.55383L6.2168 10.6184L16.8116 11.3323L32.2819 10.7969L34.9073 0L27.6937 7.64187Z" fill="#935435"/>
    <path fill-rule="evenodd" clip-rule="evenodd" d="M15.1181 7.55498L28.3867 6.70293L15.5006 6.29584L6.2168 6.56881L15.1181 7.55498Z" fill="#42210B"/>
    <path fill-rule="evenodd" clip-rule="evenodd" d="M11.2747 0C10.678 0 10.1125 0.266434 9.73257 0.726544L4.43899 7.13716C3.88351 7.80985 4.39556 8.82157 5.26657 8.77229L16.3076 8.14762L29.1389 8.84007C29.8449 8.87817 30.5186 8.54063 30.9108 7.95229L36.2118 0H11.2747Z" fill="#FFCD5E"/>
</svg>
