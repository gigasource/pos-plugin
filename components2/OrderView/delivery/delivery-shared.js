import {ref} from "vue";

export const deliveryOrderMode = ref('tablet');
export const showKeyboard = ref(false); // only show in mobile mode
export const favorites = ref([]);
export const selectedCustomer = ref({});
export const selectedAddress = ref(0);

export const name = ref('');
export const phone = ref('');
export const address = ref('');
export const zipcode = ref('');
export const street = ref('')
export const house = ref('')
export const city = ref('')
export const dialog = ref({
  input: false,
  order: false,
  choice: false,
  note: false,
});

export const placeId = ref('');

export const dialogMode = ref('add')

export function openDialog(mode, _address, _zipcode, index) {
  if (mode === 'add') {
    name.value = selectedCustomer.value.name || ''
    phone.value = selectedCustomer.value.phone || ''
    address.value = ''
    zipcode.value = ''
    house.value = ''
    street.value = ''
    city.value = ''
    placeId.value = ''
  } else if (mode === 'edit') {
    name.value = selectedCustomer.value.name || ''
    address.value = _address
    zipcode.value = _zipcode
    selectedAddress.value = index
  }
  dialogMode.value = mode
  dialog.value.input = true
}

export const autocompleteAddresses = ref([]);
