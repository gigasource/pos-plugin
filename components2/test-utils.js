import "../jest.setup";
import _ from 'lodash';
import {config, mount} from "@vue/test-utils";
import {hooks} from "./OrderView/pos-logic";
import {disablePay, payBtnClickable, payPrintMode, showIcon} from "./OrderView/pos-logic-be";
import StubFactory from './StubFactory';

jest.mock("vue-i18n", () => {
  return {
    useI18n() {
      return {
        t(t) {
          return t; // [Consider] return `t('${t}')` to indicate that i18n has been used
        },
        locale: "en"
      };
    }
  };
});
jest.mock("vue-router", () => {
  return {
    useRoute() {
      return {
        query: {
          type: 'default'
        }
      };
    },
    useRouter() {
      return {
        push() {
        }
      };
    }
  };
});

const dayjs = require("dayjs");
export const orm = require("schemahandler");
const syncFlow = require("schemahandler/sync/sync-flow");
const syncPlugin = require("schemahandler/sync/sync-plugin-multi");
const {stringify} = require("schemahandler/utils");
//jest.useFakeTimers("modern").setSystemTime(new Date("2021-01-01").getTime());
require("mockdate").set(new Date("2021-01-01").getTime());

const delay = require("delay");

export const foodTax = {taxes: [5, 10]};
export const drinkTax = {taxes: [16, 32]};

export const cola = {name: "Cola", price: 1.3, quantity: 1, ...drinkTax};
export const fanta = {name: "Fanta", price: 2, quantity: 1, ...drinkTax};
export const rice = {name: "Rice", price: 10, quantity: 1, ...foodTax};
export const ketchup = {name: "Add Ketchup", price: 3, quantity: 1};

beforeAll(async () => {
  orm.connect({uri: "mongodb://localhost:27017"}, "myproject");
  orm.registerSchema("Order", {
    inProgress: Boolean,
    items: [{}],
    table: Number
  });
  await orm("Order").deleteMany();
  await orm("Commit").deleteMany();

  orm.plugin(syncPlugin);
  orm.plugin(syncFlow);
  orm.registerCommitBaseCollection("Order");
  orm.plugin(require("../backend/commit/orderCommit"));
  orm.registerCollectionOptions("Order");
  orm.emit("commit:flow:setMaster", true);
});

afterEach(async () => {
  await orm("Order")
    .deleteMany()
    .direct();
  await orm("Commit")
    .deleteMany()
    .direct();
});

export let actions = [];
["showOrderReceipt", "pay", "printOrder"].forEach(e =>
  hooks.on(e, () => actions.push(e))
);
export let expectArray = () => [
  "payBtnClickable",
  payBtnClickable.value,
  "payPrintMode",
  payPrintMode.value,
  "showIcon",
  showIcon.value,
  "disablePay",
  disablePay.value,
  actions
];
//</editor-fold>

//<editor-fold desc="component">
export let component, wrapper;

export function setComponent(_component) {
  component = _component;
}

/**
 *
 * @param _component
 * @param options
 * @param useDefaults indicate whether _.defaults will be use for options
 * @returns {*}
 * example: makeWrapper(Order2, {shallow: false});
 */
export const makeWrapper = (_component, options, useDefaults = false) => {
  const defaultOption = {
    props: {},
    shallow: true,
    global: {
      directives: {
        touch: true
      },
      stubs: {
        'choose-table-dialog': true,
        'dialog-change-value': true,
        'dialog-choose-popup-modifier': true,
        "dialog-config-order-item": true,
        'dialog-form-input': true,
        'dialog-multi-payment': true,
        'dialog-multi-payment2': true,
        'dialog-product-search-result': true,
        'dialog-text-filter': true,
        'g-avatar': true,
        'g-autocomplete': true,
        'g-badge': true,
        'g-btn': true,
        "g-btn-bs": true,
        'g-card': true,
        'g-card-title': true,
        'g-combobox': true,
        'g-dialog': true,
        'g-divider': true,
        "g-icon": true,
        'g-item-group': true,
        'g-menu': true,
        "g-overlay": true,
        'g-radio': true,
        'g-simple-table': true,
        'g-spacer': true,
        'g-switch': true,
        'g-table': true,
        'g-tabs': true,
        'g-tab-item': true,
        'g-text-field': true,
        'g-text-field-bs': true,
        'g-toolbar': true,
        'g-snackbar': true,
        'pos-order': true,
        'pos-quick-order-toolbar': true,
        //'pos-order-split-order': true,
        'pos-keyboard-full': true,
        'pos-order-keyboard': true,
        'pos-order-move-items': true,
        'pos-order-receipt': true,
        'pos-order-voucher-dialog': true,
        'pos-textfield-new': true,
        'g-grid-select': StubFactory('g-grid-select'),
        'pos-payment-screen-payment-methods2': true,
        'pos-payment-screen-keyboard2': true,
        'pos-restaurant-payment-order-detail2': true,
        'pos-restaurant-payment-toolbar2': true,
      },
      mocks: {
        t: a => a,
        $t: a => a,
        $filters: {
          formatCurrency(val, decimals = 2) {
            if (!val || isNaN(val) || Math.floor(val) === val) return val;
            return val.toFixed(decimals);
          }
        }
      }
    }
  }
  const optionResult = useDefaults ? _.defaults(options, defaultOption) : _.merge(defaultOption, options)
  wrapper = mount(component || _component, optionResult);
};

beforeAll(() => (config.renderStubDefaultSlot = true));
