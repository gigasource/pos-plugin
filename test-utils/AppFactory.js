export default function (component) {
	return {
		name: 'App',
		components: {
			Component: component
		},
		render() {
			return <Component/>
		}
	}
}
