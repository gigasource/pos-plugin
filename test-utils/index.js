const cmsFactory = require('./mockCms')
const syncFactory = require('./mockSync')

module.exports = {
	cmsFactory,
	syncFactory
}
