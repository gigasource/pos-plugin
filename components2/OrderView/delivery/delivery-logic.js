import { computed, ref } from 'vue';
import cms from 'cms';
import { favorites } from './delivery-shared';

export const currentOrder = ref({
  items: [],
  hasOrderWideDiscount: false,
  firstInit: false
})
export const itemsWithQty = computed(() => {
  if (currentOrder.value && currentOrder.value.items)
    return currentOrder.value.items.filter(i => i.quantity > 0)
  return []
});

// inc 1
export function addItem(item) {
  //fixme: refactore
}

// dec 1
export function removeItem(item) {
  //fixme: refactore
}

// remove modifier from specified item in order
export function removeModifier(item, index) {
  //fixme: refactore
}

// online-order products
export const products = ref([])
export async function loadProduct() {
  products.value = (await cms.getModel('Product').find({type: 'delivery'})).map(p => ({ ...p, text: `${p.id}. ${p.name}` }))
  favorites.value = products.value.filter(p => p.option.favorite)
}

export const keyboardConfig = ref([])
export async function loadKeyboard() {
  const setting = await cms.getModel('PosSetting').findOne()
  keyboardConfig.value = setting && setting['keyboardDeliveryConfig']
}
