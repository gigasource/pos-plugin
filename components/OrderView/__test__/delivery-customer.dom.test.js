import {makeWrapper, wrapper} from "../../test-utils";
import _ from 'lodash'
const delay = require("delay");
import {
  deliveryOrderMode, favorites, openDialog, selectedCustomer, showKeyboard,
  name, phone, address, zipcode, street, house, city, selectedAddress, placeId, autocompleteAddresses
} from "../delivery/delivery-shared";
import { deliveryCustomerUiFactory } from '../delivery/delivery-customer-ui'
import { dialogMode } from '../delivery/delivery-shared';

import defaultMdc, { mockMissedCalls, mockFavorites, selectedCustomerMock } from './mock_delivery_data';
import { nextTick } from 'vue';
const runTestWithConfig = (mdc) => {
  mdc = _.merge(defaultMdc, mdc)
  deliveryOrderMode.value = mdc.deliveryOrderMode
  showKeyboard.value = mdc.showKeyboard
  favorites.value = mdc.favorites
  selectedCustomer.value = mdc.selectedCustomer
  selectedAddress.value = mdc.selectedAddress
  name.value = mdc.name
  phone.value = mdc.phone
  address.value = mdc.address
  zipcode.value = mdc.zipcode
  street.value = mdc.street
  house.value = mdc.house
  city.value = mdc.city
  placeId.value = mdc.placeId
  dialogMode.value = mdc.dialogMode
  autocompleteAddresses.value = mdc.autocompleteAddresses

  const Root = {
    name: 'CustomerUIRender',
    setup() {
      const { customerUiRender, calls, missedCalls, orderType } =  deliveryCustomerUiFactory()
      calls.value = mdc.calls || [] // default value
      missedCalls.value = mdc.missedCalls || [] // default value
      orderType.value = mdc.orderType // null || pickup || delivery
      return customerUiRender
    }
  }

  makeWrapper(Root)

  expect(wrapper.html()).toMatchSnapshot()
}

describe('order-delivery', () => {
  describe('favorites', () => {
    async function runRenderFavoriteTest() {
      favorites.value = mockFavorites
      makeWrapper({
        name: 'CustomerUIRender',
        setup() {
          const { renderFavorites } =  deliveryCustomerUiFactory()
          return renderFavorites
        }
      })
      await nextTick()
      await delay(50)
      expect(wrapper.html()).toMatchSnapshot()
    }
    it('should not render favorites in mobile mode when keyboard show', async () => {
      deliveryOrderMode.value = 'mobile'
      showKeyboard.value = true
      await runRenderFavoriteTest()
    })
    it("should render favorites in mobile mode when keyboard hide", async () => {

      deliveryOrderMode.value = 'mobile'
      showKeyboard.value = false
      await runRenderFavoriteTest()
    })
    it('should render favorites in tablet mode', async () => {
      deliveryOrderMode.value = 'tablet'
      showKeyboard.value = true
      await runRenderFavoriteTest()
    })
  })

  describe('selected customer', () => {
    async function runTest() {
      makeWrapper({
        name: 'CustomerUIRender',
        setup() {
          const { renderSelectedCustomer } =  deliveryCustomerUiFactory()
          return renderSelectedCustomer
        }
      })
      await nextTick()
      await delay(50)
      expect(wrapper.html()).toMatchSnapshot()
    }

    it('should render selected customer with selected address is the 2nd address', async () => {
      selectedCustomer.value = selectedCustomerMock
      selectedAddress.value = 1
      await runTest()
    })
  })

  describe('in mobile', function () {
    it('should render correct for new customer', () => {
      runTestWithConfig({
        deliveryOrderMode: 'mobile',
        selectedCustomer: null
      })
    })
    it('should render correct for selected customer', () => {
      // see selected customer in mock data
      runTestWithConfig({
        deliveryOrderMode: 'mobile'
      })
    })
    it('should show keyboard', () => {
      runTestWithConfig({
        deliveryOrderMode: 'mobile',
        showKeyboard: true
      })
    })
  });

  describe('in tablet', () => {
    it('should render correct for new customer', () => {
      runTestWithConfig({
        deliveryOrderMode: 'tablet',
        selectedCustomer: null
      })
    })
    it('should render correct for selected customer', () => {
      runTestWithConfig({
        deliveryOrderMode: 'tablet'
      })
    })
  })

  describe('missed calls', () => {
    it('should render missed calls', () => {
      runTestWithConfig({
        missedCalls: mockMissedCalls
      })
    })
  })
})
