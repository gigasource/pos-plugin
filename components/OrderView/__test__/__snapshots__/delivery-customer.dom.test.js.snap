// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`order-delivery favorites should render favorites in mobile mode but does't show keyboard 1`] = `
<div class="delivery-info__favorite">
</div>
`;

exports[`order-delivery favorites should render favorites in mobile mode when keyboard hide 1`] = `
<div class="delivery-info__favorite">
  <div style="background-color: rgb(251, 228, 236);"
       class="delivery-info__favorite-item"
  >
    Cola
  </div>
  <div style="background-color: rgb(132, 255, 255);"
       class="delivery-info__favorite-item"
  >
    Pepsi
  </div>
  <div style="background-color: rgb(128, 216, 255);"
       class="delivery-info__favorite-item"
  >
    Water
  </div>
  <div style="background-color: rgb(255, 245, 157);"
       class="delivery-info__favorite-item"
  >
    Baijiu
  </div>
  <div style="background-color: rgb(178, 255, 89);"
       class="delivery-info__favorite-item"
  >
    Whiskey
  </div>
  <div style="background-color: rgb(225, 190, 231);"
       class="delivery-info__favorite-item"
  >
    Pudding
  </div>
  <div style="background-color: rgb(255, 171, 145);"
       class="delivery-info__favorite-item"
  >
    Olive
  </div>
</div>
`;

exports[`order-delivery favorites should render favorites in tablet 1`] = `
<div class="delivery-info__favorite">
  <div style="background-color: rgb(251, 228, 236);"
       class="delivery-info__favorite-item"
  >
    Cola
  </div>
  <div style="background-color: rgb(132, 255, 255);"
       class="delivery-info__favorite-item"
  >
    Pepsi
  </div>
  <div style="background-color: rgb(128, 216, 255);"
       class="delivery-info__favorite-item"
  >
    Water
  </div>
  <div style="background-color: rgb(255, 245, 157);"
       class="delivery-info__favorite-item"
  >
    Baijiu
  </div>
  <div style="background-color: rgb(178, 255, 89);"
       class="delivery-info__favorite-item"
  >
    Whiskey
  </div>
  <div style="background-color: rgb(225, 190, 231);"
       class="delivery-info__favorite-item"
  >
    Pudding
  </div>
  <div style="background-color: rgb(255, 171, 145);"
       class="delivery-info__favorite-item"
  >
    Olive
  </div>
</div>
`;

exports[`order-delivery in mobile should render correct for new customer 1`] = `
<div class="delivery-info">
  <div class="delivery-info--upper">
    <div class="delivery-info__favorite">
      <div style="background-color: rgb(251, 228, 236);"
           class="delivery-info__favorite-item"
      >
        Cola
      </div>
      <div style="background-color: rgb(132, 255, 255);"
           class="delivery-info__favorite-item"
      >
        Pepsi
      </div>
      <div style="background-color: rgb(128, 216, 255);"
           class="delivery-info__favorite-item"
      >
        Water
      </div>
      <div style="background-color: rgb(255, 245, 157);"
           class="delivery-info__favorite-item"
      >
        Baijiu
      </div>
      <div style="background-color: rgb(178, 255, 89);"
           class="delivery-info__favorite-item"
      >
        Whiskey
      </div>
      <div style="background-color: rgb(225, 190, 231);"
           class="delivery-info__favorite-item"
      >
        Pudding
      </div>
      <div style="background-color: rgb(255, 171, 145);"
           class="delivery-info__favorite-item"
      >
        Olive
      </div>
    </div>
    <div class="delivery-info__customer">
      <div class="row-flex mt-3 w-100">
        <div style="flex: 1; margin-right: 2px;">
          <g-text-field-stub outlined="true"
                             dense="true"
                             modelvalue="084.999.8888"
                             label="Phone"
          >
          </g-text-field-stub>
        </div>
        <div style="flex: 1; margin-left: 2px;">
          <g-text-field-stub outlined="true"
                             dense="true"
                             modelvalue="[object Object]"
                             label="Name"
          >
          </g-text-field-stub>
        </div>
      </div>
      <div class="row-flex">
        <div class="col-9">
          <g-combobox-stub style="width: 100%;"
                           label="Address"
                           modelvalue
                           outlined="true"
                           dense="true"
                           clearable="true"
                           skip-search="true"
                           items
                           keep-menu-on-blur="true"
                           menu-class="menu-autocomplete-address"
          >
          </g-combobox-stub>
        </div>
        <div class="flex-grow-1 ml-1">
          <g-text-field-stub outlined="true"
                             dense="true"
                             modelvalue
                             label="Nr"
          >
          </g-text-field-stub>
        </div>
      </div>
    </div>
  </div>
  <div class="delivery-info--lower">
    <div class="delivery-info__call--empty">
      <p class="fw-700">
        Empty
      </p>
      <p class="text-grey-darken-1">
        No pending calls
      </p>
    </div>
  </div>
</div>
`;

exports[`order-delivery in mobile should render correct for selected customer 1`] = `
<div class="delivery-info">
  <div class="delivery-info--upper">
    <div class="delivery-info__favorite">
      <div style="background-color: rgb(251, 228, 236);"
           class="delivery-info__favorite-item"
      >
        Cola
      </div>
      <div style="background-color: rgb(132, 255, 255);"
           class="delivery-info__favorite-item"
      >
        Pepsi
      </div>
      <div style="background-color: rgb(128, 216, 255);"
           class="delivery-info__favorite-item"
      >
        Water
      </div>
      <div style="background-color: rgb(255, 245, 157);"
           class="delivery-info__favorite-item"
      >
        Baijiu
      </div>
      <div style="background-color: rgb(178, 255, 89);"
           class="delivery-info__favorite-item"
      >
        Whiskey
      </div>
      <div style="background-color: rgb(225, 190, 231);"
           class="delivery-info__favorite-item"
      >
        Pudding
      </div>
      <div style="background-color: rgb(255, 171, 145);"
           class="delivery-info__favorite-item"
      >
        Olive
      </div>
    </div>
    <div class="delivery-info__customer">
      <div class="row-flex mt-3 w-100">
        <div style="flex: 1; margin-right: 2px;">
          <g-text-field-stub outlined="true"
                             dense="true"
                             modelvalue="084.999.8888"
                             label="Phone"
          >
          </g-text-field-stub>
        </div>
        <div style="flex: 1; margin-left: 2px;">
          <g-text-field-stub outlined="true"
                             dense="true"
                             modelvalue="[object Object]"
                             label="Name"
          >
          </g-text-field-stub>
        </div>
      </div>
      <div class="row-flex">
        <div class="col-9">
          <g-combobox-stub style="width: 100%;"
                           label="Address"
                           modelvalue
                           outlined="true"
                           dense="true"
                           clearable="true"
                           skip-search="true"
                           items
                           keep-menu-on-blur="true"
                           menu-class="menu-autocomplete-address"
          >
          </g-combobox-stub>
        </div>
        <div class="flex-grow-1 ml-1">
          <g-text-field-stub outlined="true"
                             dense="true"
                             modelvalue
                             label="Nr"
          >
          </g-text-field-stub>
        </div>
      </div>
    </div>
  </div>
  <div class="delivery-info--lower">
    <div class="delivery-info__call--empty">
      <p class="fw-700">
        Empty
      </p>
      <p class="text-grey-darken-1">
        No pending calls
      </p>
    </div>
  </div>
</div>
`;

exports[`order-delivery in mobile should show keyboard 1`] = `
<div class="delivery-info">
  <div class="delivery-info--upper">
    <div class="delivery-info__customer">
      <div class="row-flex mt-3 w-100">
        <div style="flex: 1; margin-right: 2px;">
          <g-text-field-stub outlined="true"
                             dense="true"
                             modelvalue="084.999.8888"
                             label="Phone"
          >
          </g-text-field-stub>
        </div>
        <div style="flex: 1; margin-left: 2px;">
          <g-text-field-stub outlined="true"
                             dense="true"
                             modelvalue="[object Object]"
                             label="Name"
          >
          </g-text-field-stub>
        </div>
      </div>
      <div class="row-flex">
        <div class="col-9">
          <g-combobox-stub style="width: 100%;"
                           label="Address"
                           modelvalue
                           outlined="true"
                           dense="true"
                           clearable="true"
                           skip-search="true"
                           items
                           keep-menu-on-blur="true"
                           menu-class="menu-autocomplete-address"
          >
          </g-combobox-stub>
        </div>
        <div class="flex-grow-1 ml-1">
          <g-text-field-stub outlined="true"
                             dense="true"
                             modelvalue
                             label="Nr"
          >
          </g-text-field-stub>
        </div>
      </div>
    </div>
  </div>
  <div class="delivery-info--lower">
    <div class="delivery-info__call--empty">
      <p class="fw-700">
        Empty
      </p>
      <p class="text-grey-darken-1">
        No pending calls
      </p>
    </div>
  </div>
</div>
`;

exports[`order-delivery in tablet should render correct for new customer 1`] = `
<div class="delivery-info">
  <div class="delivery-info--upper">
    <div class="delivery-info__favorite">
      <div style="background-color: rgb(251, 228, 236);"
           class="delivery-info__favorite-item"
      >
        Cola
      </div>
      <div style="background-color: rgb(132, 255, 255);"
           class="delivery-info__favorite-item"
      >
        Pepsi
      </div>
      <div style="background-color: rgb(128, 216, 255);"
           class="delivery-info__favorite-item"
      >
        Water
      </div>
      <div style="background-color: rgb(255, 245, 157);"
           class="delivery-info__favorite-item"
      >
        Baijiu
      </div>
      <div style="background-color: rgb(178, 255, 89);"
           class="delivery-info__favorite-item"
      >
        Whiskey
      </div>
      <div style="background-color: rgb(225, 190, 231);"
           class="delivery-info__favorite-item"
      >
        Pudding
      </div>
      <div style="background-color: rgb(255, 171, 145);"
           class="delivery-info__favorite-item"
      >
        Olive
      </div>
    </div>
    <div class="delivery-info__customer">
      <g-text-field-bs-stub class="bs-tf__pos"
                            label="Name"
      >
      </g-text-field-bs-stub>
      <g-text-field-bs-stub class="bs-tf__pos"
                            label="Address"
                            modelvalue
      >
      </g-text-field-bs-stub>
    </div>
  </div>
  <div class="delivery-info--lower">
    <div class="delivery-info__call--empty">
      <p class="fw-700">
        Empty
      </p>
      <p class="text-grey-darken-1">
        No pending calls
      </p>
    </div>
  </div>
</div>
`;

exports[`order-delivery in tablet should render correct for selected customer 1`] = `
<div class="delivery-info">
  <div class="delivery-info--upper">
    <div class="delivery-info__favorite">
      <div style="background-color: rgb(251, 228, 236);"
           class="delivery-info__favorite-item"
      >
        Cola
      </div>
      <div style="background-color: rgb(132, 255, 255);"
           class="delivery-info__favorite-item"
      >
        Pepsi
      </div>
      <div style="background-color: rgb(128, 216, 255);"
           class="delivery-info__favorite-item"
      >
        Water
      </div>
      <div style="background-color: rgb(255, 245, 157);"
           class="delivery-info__favorite-item"
      >
        Baijiu
      </div>
      <div style="background-color: rgb(178, 255, 89);"
           class="delivery-info__favorite-item"
      >
        Whiskey
      </div>
      <div style="background-color: rgb(225, 190, 231);"
           class="delivery-info__favorite-item"
      >
        Pudding
      </div>
      <div style="background-color: rgb(255, 171, 145);"
           class="delivery-info__favorite-item"
      >
        Olive
      </div>
    </div>
    <div class="delivery-info__customer">
      <g-text-field-bs-stub class="bs-tf__pos"
                            label="Name"
      >
      </g-text-field-bs-stub>
      <g-text-field-bs-stub class="bs-tf__pos"
                            label="Address"
                            modelvalue
      >
      </g-text-field-bs-stub>
    </div>
  </div>
  <div class="delivery-info--lower">
    <div class="delivery-info__call--empty">
      <p class="fw-700">
        Empty
      </p>
      <p class="text-grey-darken-1">
        No pending calls
      </p>
    </div>
  </div>
</div>
`;

exports[`order-delivery missed calls should render missed calls 1`] = `
<div class="delivery-info">
  <div class="delivery-info--upper">
    <div class="delivery-info__favorite">
      <div style="background-color: rgb(251, 228, 236);"
           class="delivery-info__favorite-item"
      >
        Cola
      </div>
      <div style="background-color: rgb(132, 255, 255);"
           class="delivery-info__favorite-item"
      >
        Pepsi
      </div>
      <div style="background-color: rgb(128, 216, 255);"
           class="delivery-info__favorite-item"
      >
        Water
      </div>
      <div style="background-color: rgb(255, 245, 157);"
           class="delivery-info__favorite-item"
      >
        Baijiu
      </div>
      <div style="background-color: rgb(178, 255, 89);"
           class="delivery-info__favorite-item"
      >
        Whiskey
      </div>
      <div style="background-color: rgb(225, 190, 231);"
           class="delivery-info__favorite-item"
      >
        Pudding
      </div>
      <div style="background-color: rgb(255, 171, 145);"
           class="delivery-info__favorite-item"
      >
        Olive
      </div>
    </div>
    <div class="delivery-info__customer">
      <g-text-field-bs-stub class="bs-tf__pos"
                            label="Name"
      >
      </g-text-field-bs-stub>
      <g-text-field-bs-stub class="bs-tf__pos"
                            label="Address"
                            modelvalue
      >
      </g-text-field-bs-stub>
    </div>
  </div>
  <div class="delivery-info--lower">
    <div class="delivery-info__call--empty">
      <p class="fw-700">
        Empty
      </p>
      <p class="text-grey-darken-1">
        No pending calls
      </p>
    </div>
    <g-menu-stub modelvalue="false"
                 top="true"
                 left="true"
                 nudge-top="5"
    >
      <div class="menu-missed">
        <div class="menu-missed__call">
          <div class="menu-missed__call--info">
            <p class="fw-700 fs-small">
              <g-icon-stub size="16"
                           class="mr-1"
              >
                icon-call
              </g-icon-stub>
              0123456678
            </p>
            <p class="fs-small text-grey-darken-1">
              Miss Customer 1
            </p>
          </div>
          <div class="delivery-info__call-btn">
            <g-icon-stub size="20">
              icon-take-away
            </g-icon-stub>
          </div>
          <div class="delivery-info__call-btn">
            <g-icon-stub size="20">
              icon-delivery-scooter
            </g-icon-stub>
          </div>
          <div class="delivery-info__call-btn--cancel">
            <g-icon-stub color="white">
              clear
            </g-icon-stub>
          </div>
        </div>
        <div class="menu-missed__call">
          <div class="menu-missed__call--info">
            <p class="fw-700 fs-small">
              <g-icon-stub size="16"
                           class="mr-1"
              >
                icon-call
              </g-icon-stub>
              0123456678
            </p>
            <p class="fs-small text-grey-darken-1">
              Miss Customer 2
            </p>
          </div>
          <div class="delivery-info__call-btn">
            <g-icon-stub size="20">
              icon-take-away
            </g-icon-stub>
          </div>
          <div class="delivery-info__call-btn">
            <g-icon-stub size="20">
              icon-delivery-scooter
            </g-icon-stub>
          </div>
          <div class="delivery-info__call-btn--cancel">
            <g-icon-stub color="white">
              clear
            </g-icon-stub>
          </div>
        </div>
        <div class="menu-missed__call">
          <div class="menu-missed__call--info">
            <p class="fw-700 fs-small">
              <g-icon-stub size="16"
                           class="mr-1"
              >
                icon-call
              </g-icon-stub>
              0123456678
            </p>
            <p class="fs-small text-grey-darken-1">
              Miss Customer 3
            </p>
          </div>
          <div class="delivery-info__call-btn">
            <g-icon-stub size="20">
              icon-take-away
            </g-icon-stub>
          </div>
          <div class="delivery-info__call-btn">
            <g-icon-stub size="20">
              icon-delivery-scooter
            </g-icon-stub>
          </div>
          <div class="delivery-info__call-btn--cancel">
            <g-icon-stub color="white">
              clear
            </g-icon-stub>
          </div>
        </div>
      </div>
    </g-menu-stub>
  </div>
</div>
`;

exports[`order-delivery selected customer should render selected customer with selected address is the 2nd address 1`] = `
<div class="delivery-info__customer-address">
  <div class="row-flex align-items-center">
    <g-radio-stub small="true"
                  modelvalue="1"
                  value="0"
                  label="Address 1"
                  color="#536DFE"
    >
    </g-radio-stub>
    <g-spacer-stub>
    </g-spacer-stub>
    <g-btn-bs-stub small="true"
                   style="margin: 0px 2px; padding: 4px;"
                   background-color="#F9A825"
    >
      <g-icon-stub size="15">
        icon-reservation_modify
      </g-icon-stub>
    </g-btn-bs-stub>
    <g-btn-bs-stub small="true"
                   style="margin: 0px 2px; padding: 4px;"
                   background-color="#FF4452"
    >
      <g-icon-stub size="15">
        icon-delete
      </g-icon-stub>
    </g-btn-bs-stub>
  </div>
  <p>
    address val 1
  </p>
  <p class="text-grey fs-small">
    zipcode val 1
  </p>
  <p class="text-grey fs-small">
    city val 1
  </p>
</div>
<div class="delivery-info__customer-address delivery-info__customer-address--selected">
  <div class="row-flex align-items-center">
    <g-radio-stub small="true"
                  modelvalue="1"
                  value="1"
                  label="Address 2"
                  color="#536DFE"
    >
    </g-radio-stub>
    <g-spacer-stub>
    </g-spacer-stub>
    <g-btn-bs-stub small="true"
                   style="margin: 0px 2px; padding: 4px;"
                   background-color="#F9A825"
    >
      <g-icon-stub size="15">
        icon-reservation_modify
      </g-icon-stub>
    </g-btn-bs-stub>
    <g-btn-bs-stub small="true"
                   style="margin: 0px 2px; padding: 4px;"
                   background-color="#FF4452"
    >
      <g-icon-stub size="15">
        icon-delete
      </g-icon-stub>
    </g-btn-bs-stub>
  </div>
  <p>
    address val 2
  </p>
  <p class="text-grey fs-small">
    zipcode val 2
  </p>
  <p class="text-grey fs-small">
    city val 2
  </p>
</div>
<g-icon-stub size="40"
             color="#1271FF"
>
  add_circle
</g-icon-stub>
`;
